### PR TITLE
office

### PR DESCRIPTION
**Pull request by KhusanKholmatov**

### CODE REVIEW

**Review by gargatuma — DISMISSED** [review]
good job)

File: src/main/java/com/exadel/coolDesking/office/OfficeCreateDto.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class OfficeCreateDto {
+    private String name;
+    private String country;
+    private String city;
+    private String address;
+    @JsonProperty("parking_capacity")
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/office/OfficeCreateDto.java:17]: remove it. `parkingCapacity` is absolutely fine

File: src/main/java/com/exadel/coolDesking/office/OfficeService.java [hunk 2]
```diff
@@ -1,21 +1,41 @@
...
+        return officeRepository.findAll();
+    }
 
-    public Office create(Office office) {
-        return null;
+    public OfficeAddedResponse create(OfficeCreateDto officeCreateDto) {
+        if (officeRepository.existsByName(officeCreateDto.getName())){
+            throw new ConflictException("Office with this name already exists", Office.class, officeCreateDto.getName());
+        }
+        return new OfficeAddedResponse(officeRepository.save(officeMapper.officeDtoToOfficeEntity(officeCreateDto)).getId());
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/office/OfficeService.java:27]: create a new varieble for `officeRepository.save(officeMapper.officeDtoToOfficeEntity(officeCreateDto))` - it'll be easier to read

File: src/main/java/com/exadel/coolDesking/office/OfficeResponse.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.util.List;
+import java.util.UUID;
+
+@Setter
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor
+public class OfficeResponse {
+    private Office office;
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/office/OfficeResponse.java:17]: No need to use it as a field. for now you can just `extend` your `OfficeResponse` from `Office`

**Review by ohalushchak-exadel — CHANGES_REQUESTED** [review]

**Pull request closed without merging**